virtualmachine/vsphere/vm.go

```diff
@@ -539,35 +540,42 @@ func (vm *VM) RemoveDisk(vmdkName string) (err error) {
 		return fmt.Errorf("Failed to retrieve datacenter: %s", err)
 	}
 
-	// finds the virtualmachine with name vm.Name
-	vmMo, err := findVM(vm, dcMo, vm.Name)
-	if err != nil {
-		return fmt.Errorf("VM not found", vm.Name, err)
-	}
-
-	// find the virtual disk to be removed from the vm
-	var deviceMo *types.VirtualDisk
-	for _, d := range vmMo.Config.Hardware.Device {
-		switch device := d.(type) {
-		case *types.VirtualDisk:
-			fileName := d.GetVirtualDevice().Backing.(types.BaseVirtualDeviceFileBackingInfo).GetVirtualDeviceFileBackingInfo().FileName
-			if strings.HasSuffix(fileName, vmdkName) {
-				deviceMo = device
-				break
+	errorMessage = ""
```

Comment:
not required. 
It is already initialized with zero value, when declare without explicit initialization.